osu.Desktop/Admin/AdminChecker.cs

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Security.Principal;
+using osu.Framework;
+using osu.Framework.Allocation;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Sprites;
+using osu.Game.Graphics;
+using osu.Game.Overlays;
+using osu.Game.Overlays.Notifications;
+
+namespace osu.Desktop.Admin
+{
+    /// <summary>
+    /// Checks if the game is running with elevated privileges (as admin in Windows, root in Unix) and displays a warning notification if so.
+    /// </summary>
+    public class AdminChecker : CompositeDrawable
+    {
+        [Resolved]
+        protected NotificationOverlay Notifications { get; private set; }
+
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+            if (isAdmin())
+                Notifications.Post(new AdminNotification());
+        }
+
+        private bool isAdmin() => OperatingSystem.IsWindows() ? new WindowsPrincipal(WindowsIdentity.GetCurrent()).IsInRole(WindowsBuiltInRole.Administrator) : Mono.Unix.Native.Syscall.geteuid() == 0;
+
+        private class AdminNotification : SimpleNotification
```

Comment:
`ElevatedPrivilagesNotification`

Reply:
Should I rename `isAdmin()` as well?

Reply:
yeah

Reply:
Wanted to clarify: Should I have a separate `isElevated()` function (that's not one line) or have it inlined? I need `OperatingSystem.IsWindows()` to avoid CA1416 so that's why I didn't use a switch statement.

```cs
if (OperatingSystem.IsWindows() ? new WindowsPrincipal(WindowsIdentity.GetCurrent()).IsInRole(WindowsBuiltInRole.Administrator) : Mono.Unix.Native.Syscall.geteuid() == 0)
    Notifications.Post(new ElevatedPrivilegesNotification());
```

Reply:
> I need OperatingSystem.IsWindows() to avoid CA1416 so that's why I didn't use a switch statement.

right, seems that is also used elsewhere for the same reason, fair enough.

> Should I have a separate isElevated() function (that's not one line) or have it inlined?

like this:
```csharp

bool elevated = false;

if (OperatingSystem.IsWindows())
{
    var windowsIdentity = WindowsIdentity.GetCurrent();
    var windowsPrincipal = new WindowsPrincipal(windowsIdentity);
    elevated = windowsPrincipal.IsInRole(WindowsBuiltInRole.Administrator);
}
else if (RuntimeInfo.IsUnix)
{
    elevated = Mono.Unix.Native.Syscall.geteuid() == 0;
}

if (!elevated)
    return;
    
// post elevated privileges notification.
```